web/angular-wallet/src/app/main/messages/message-view/message-view.component.html

```diff
@@ -81,15 +81,16 @@
 
         <div class="bubble">
           <div class="message encrypted" *ngIf="message.encryptedMessage && !message.message">
-            <div *ngIf="!showPin">
-              <mat-icon class="s-16 lock" (click)="showPin=true">lock</mat-icon>
+            <div *ngIf="!showPin[message.timestamp]">
+              <mat-icon class="s-16 lock" (click)="showPin[message.timestamp]=true">lock</mat-icon>
```

Comment:
Only show one PIN form - it was weird showing a bunch of PIN forms...

Reply:
:clap: :clap: :clap: